packages/pyodide-kernel/src/worker.ts

```diff
@@ -1,509 +1,579 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
+import coincident from 'coincident';
+
 import type Pyodide from 'pyodide';
 
-import type { DriveFS } from '@jupyterlite/contents';
+import {
+  ContentsAPI,
+  DriveFS,
+  TDriveRequest,
+  TDriveMethod,
+  TDriveResponse,
+  ServiceWorkerContentsAPI,
+} from '@jupyterlite/contents';
 
 import type { IPyodideWorkerKernel } from './tokens';
+import { PyodideInterface } from 'pyodide';
 
-export class PyodideRemoteKernel {
-  constructor() {
-    this._initialized = new Promise((resolve, reject) => {
-      this._initializer = { resolve, reject };
-    });
-  }
+declare global {
+  var ready: Promise<unknown>;
 
-  /**
-   * Accept the URLs from the host
-   **/
-  async initialize(options: IPyodideWorkerKernel.IOptions): Promise<void> {
-    this._options = options;
+  var runtime: PyodideInterface;
 
-    if (options.location.includes(':')) {
-      const parts = options.location.split(':');
-      this._driveName = parts[0];
-      this._localPath = parts[1];
-    } else {
-      this._driveName = '';
-      this._localPath = options.location;
-    }
+  var kernel: any;
```

Comment:
For the record, this is the reason of the `no-var` eslint rule removal. I'm not sure how to implement it differently.

Reply:
An alternative could be to use inline comments to disable the rules, per line or for the whole file?

Reply:
oh... this removes all of the class structure. This is not pleasing to me, as it would be all but impossible to overload by extension and add Just One Change to a relatively small method, rather than copy-pasta forking.

Reply:
> further, the any make me really sad. Is it entirely unknowable?

Those variables were already `any`, it's not a change from this PR.

> oh... this removes all of the class structure. This is not pleasing to me, as it would be all but impossible to overload by extension and add Just One Change to a relatively small method, rather than copy-pasta forking.

I know... I'm not sure of a way to expose a class instance using `coincident`, I will look further

Reply:
Doing only the coincident transition and trying to keep the class def in https://github.com/jupyterlite/pyodide-kernel/pull/115